File: Serene/Serene.Script/Northwind/Region/RegionDialog.cs

namespace Serene.Northwind
{
    using jQueryApi;
    using Serenity;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    [Imported]
    [IdProperty("RegionID"), NameProperty("RegionDescription")]
    [FormKey("Northwind.Region"), LocalTextPrefix("Northwind.Region"), Service("Northwind/Region")]
    public class RegionDialog : EntityDialog<RegionRow>
    {
    }
}

File: Serene/Serene.Web/Modules/Northwind/Region/RegionDialog.ts
namespace Serene.Northwind {

    @Serenity.Decorators.registerClass()
    @Serenity.Decorators.responsive()
    export class RegionDialog extends Serenity.EntityDialog<RegionRow, any> {
        protected getFormKey() { return RegionForm.formKey; }
        protected getIdProperty() { return RegionRow.idProperty; }
        protected getLocalTextPrefix() { return RegionRow.localTextPrefix; }
        protected getNameProperty() { return RegionRow.nameProperty; }
        protected getService() { return RegionService.baseUrl; }

        protected form = new RegionForm(this.idPrefix);

        protected getLanguages() {
            return LanguageList.getValue();
        }
    }
}